features/steps/supplier_steps.py

```diff
@@ -65,6 +55,22 @@ def step_impl(context, name, address, productsItems):
     payload = json.dumps(data)
     context.resp = requests.post(create_url, data = payload, headers = headers)
 
+@when('I list all suppliers')
```

Comment:
I guess here we should make it a press button action, which is more general:
```python
@when('I press the "{button}" button')
def step_impl(context, button):
button_id = button.lower() + '-btn'
context.driver.find_element_by_id(button_id).click()
```
Specifically, here we need to press "Search" button without any inputs.

FIY, ids of buttons are "retrieve-btn", "search-btn", "clear-btn", "create-btn", "update-btn", "delete-btn" and "recommend-btn". Refer to index.html for more info.

Following the format in sample code in Pets demo would be helpful, as it would save a lot of code if we have some more general sentences.